### ISSUE TITLE
Absolute path used in compare on git version 2.25.0.windows.1

### ISSUE THREAD

**Issue by CaselIT** [eamodio/vscode-gitlens#932]
Git version 2.25 seems to cause problems when comparing two commits and with the line history.

Steps to reproduce
After installing git 2.25.0 on windows select a commit and compare with head.
In the compare part of the left bar when clicking on a file changed an empty compare windows appears. The title is `<file xx> (deleted in <commit id>)`
My guess is that it uses the absolute path instead of the local path from the base of the repository, since clicking on `copy remove url` gives an url with the absolute path in it, like `https://gitlab.com/<repo>/blob/HEAD/c:/Users/cfede/Dev/...`

There are also other strange issues, like the line history is always empty. But I guess that this is caused by the git v2.25.

Trying with a different version of git (v.2.21) fixes the problem

Extension version: 10.2.0
VS Code version: Code 1.41.1 (26076a4de974ead31f97692a0d32f90d735645c0, 2019-12-18T14:58:56.166Z)
OS version: Windows_NT x64 10.0.18363

**Comment by CaselIT** [comment]
Edit: I've checked the logging in debug and it seems indeed that the problem is the absolute path:

```
[2020-01-15 14:57:07:890] [62] LineHistoryView.getNodeLastKnownLimit completed • 0 ms — e=LineHistoryNode(gitlens:repository(C:/Users/cfede/Dev/foo):history:line(file:///c:/Users/cfede/Dev/foo/c:/Users/cfede/Dev/foo/somefile.txt[31,40-31,40]))
[2020-01-15 14:57:07:891] [65] GitService.getRepository — e=GitUri(file:///c:/Users/cfede/Dev/foo/c:/Users/cfede/Dev/foo/somefile.txt repoPath=C:/Users/cfede/Dev/foo)
```

the wrong path `file:///c:/Users/cfede/Dev/foo/c:/Users/cfede/Dev/foo/somefile.txt` is used in many places and I guess it causes the problem

**Comment by JelianRadoev** [comment]
I can confirm its bugged for me too. File history/Line history list doesn't work after i instelled git v2.25. And if i try to open file from GitLens menu it gives error, that file is not found with weird file path.

**Comment by eamodio** [comment]
@CaselIT @JelianRadoev this looks like a behavior change in Git -- see https://github.com/git-for-windows/git/issues/2478

I have a fix for it and will release a new GitLens version soon.

**Comment by jankap** [comment]
Great news you're going to fix it. I also rolled back Git to pre 2.25 on windows and it's working as expected again. Any idea when the fix comes out? 

Thanks!

**Comment by JayEn84** [comment]
I've got the same problem with a slightly different error message:
```
[2020-01-22 11:29:42:419] [C:/dev/www/cron] Git Command failed: C:\Program Files\Git\cmd\git.exe -c core.quotepath=false -c color.ui=false -c color.status=false status --porcelain=v2 --find-renames -- c:/dev/www/cron/daily.sh • c:/dev/www/cron/daily.sh: 'c:/dev/www/cron/daily.sh' is outside repository • 38 ms
```

I don't see any problematic paths in this command, but the file is definitely not outside the repository. GitLens displays the correct history in the section "REPOSITORIES". But the sections "FILE HISTORY" and "LINE HISTORY" do nor work anymore:

![grafik](https://user-images.githubusercontent.com/4518738/72891592-da570e00-3d14-11ea-93b9-0ae07bc45fd6.png)

**Comment by miguelelvir** [comment]
Just a note. This is probably what @brentarias reported in #941

**Comment by miguelelvir** [comment]
In my case, I think the error comes from [here](https://github.com/eamodio/vscode-gitlens/blob/30e1d290e8f337d56596a030ff097eeabdd98eb7/src/git/git.ts#L458), which prepends `./` to a full file path. This is called by `gitService` [here](https://github.com/eamodio/vscode-gitlens/blob/30e1d290e8f337d56596a030ff097eeabdd98eb7/src/git/gitService.ts#L2939), which seems to normalize the path first.

**Comment by alexvy86** [comment]
I'm seeing the same as @miguelelvir , GitLens' output window reports "not a valid object name" and I see a path like `./f:/...`:
![image](https://user-images.githubusercontent.com/716334/73671952-a3b4b800-4671-11ea-8a70-5bf2ff8a31a4.png)

**Comment by Xunnamius** [comment]
Just to chime in with a slightly different symptom of the same problem. I use a mapped (network) drive exposing a Samba share hosted by my linux box. Everything was working fine up until a week or two ago. After reading this thread, I fixed it by downgrading to Git for `Windows 2.24.1.windows.2`.

**Comment by charles-dexter-ward** [comment]
I am no longer using Gitlens but VSCode's default git plugin. Waiting for a fix somewhere (GitLens? Git For Windows?)
